lib/hg-utils.coffee

```diff
@@ -242,7 +242,11 @@ class Repository
       params = []
     if !util.isArray(params)
       params = [params]
-    child = spawnSync('hg', params)
+
+    return '' unless @isCommandForRepo(params)
+
+    options = { cwd: @rootPath }
+    child = spawnSync('hg', params, options)
```

Comment:
Mercurial should take care of .hg file location since we're already telling him the repository/file path. May be something related to Windows only.

> $ hg --version
> Mercurial Distributed SCM (version 3.7.2)

Can you check if your hg is up to date?!

If updating it won't solve the problem, for now you could add a verification and if the platform is Windows, add the cwd option to spawnSync method. You also have to check if tree-view file status keeps working.

About the spawnSync performance, using an async version improves things? You're doing a great work. Feel free to open a new pull request if you need so.